src/settings/groups/GroupDetailsModel.ts

```diff
@@ -0,0 +1,318 @@
+import {
+	AdministratedGroupTypeRef,
+	CustomerTypeRef,
+	Group,
+	GroupInfo,
+	GroupInfoTypeRef,
+	GroupMemberTypeRef,
+	GroupTypeRef,
+	UserTypeRef
+} from "../../api/entities/sys/TypeRefs.js"
+import {assertNotNull, LazyLoaded, neverNull, ofClass, promiseMap} from "@tutao/tutanota-utils"
+import {EntityClient} from "../../api/common/EntityClient.js"
+import {GENERATED_MAX_ID, GENERATED_MIN_ID, isSameId} from "../../api/common/utils/EntityUtils.js"
+import {BookingItemFeatureType, GroupType, OperationType} from "../../api/common/TutanotaConstants.js"
+import {localAdminGroupInfoModel} from "../LocalAdminGroupInfoModel.js"
+import {lang} from "../../misc/LanguageViewModel.js"
+import {SelectorItemList} from "../../gui/base/DropDownSelector.js"
+import {logins} from "../../api/main/LoginController.js"
+import {Dialog, stringValidator} from "../../gui/base/Dialog.js"
+import {showProgressDialog} from "../../gui/dialogs/ProgressDialog.js"
+import {showBuyDialog} from "../../subscription/BuyDialog.js"
+import {locator} from "../../api/main/MainLocator.js"
+import {BadRequestError, NotAuthorizedError, PreconditionFailedError} from "../../api/common/error/RestError.js"
+import {compareGroupInfos, getGroupInfoDisplayName} from "../../api/common/utils/GroupUtils.js"
+import {EntityUpdateData, isUpdateForTypeRef} from "../../api/main/EventController.js"
+
+export class GroupDetailsModel {
+	groupInfo: GroupInfo
+	private readonly group: LazyLoaded<Group>
```

Comment:
I am not even sure we need to use `LazyLoaded` here still but we can keep it for now